pkg/prometheus/promcfg.go

```diff
@@ -2257,29 +2258,34 @@ func (cg *ConfigGenerator) appendEvaluationInterval(slice yaml.MapSlice, evaluat
 	return append(slice, yaml.MapItem{Key: "evaluation_interval", Value: evaluationInterval})
 }
 
+func (cg *ConfigGenerator) appendGlobalLimits(slice yaml.MapSlice, limitKey string, limit *uint64, enforcedLimit *uint64) yaml.MapSlice {
+	// Use the enforced limit if no global limit is defined to ensure that scrape jobs without an explicit limit inherit the enforced limit value.
+	if ptr.Deref(limit, 0) == 0 && ptr.Deref(enforcedLimit, 0) > 0 {
+		return cg.AppendMapItem(slice, limitKey, *enforcedLimit)
+	}
+
+	if ptr.Deref(limit, 0) > 0 {
+		return cg.AppendMapItem(slice, limitKey, *limit)
+	}
```

Comment:
It reads more easily for me. Though it brings the question of what to do when global limit > enforced limit? I suggest that we clamp down the global limit to the enforced value in this case (CEL could help later on to validate this at the API level).
```suggestion
	if ptr.Deref(limit, 0) > 0 {
		return cg.AppendMapItem(slice, limitKey, *limit)
	}

	// <explain why enforcedlimit is used>
	if ptr.Deref(enforcedLimit, 0) > 0 {
		return cg.AppendMapItem(slice, limitKey, *enforcedLimit)
	}
```